docs/faq.rst

```diff
@@ -48,6 +48,44 @@ and platform/software dependency. Combining Borg with the mechanisms provided
 by the platform (snapshots, hypervisor features) will be the best approach
 to start tackling them.
 
+How can I decrease the size of disk image backups?
+--------------------------------------------------
+
+Full disk images are as large as the full disk when uncompressed and don't get much
+smaller post-deduplication after heavy use. This is because virtually all file systems
+don't actually delete the data on disk (that is the place of so-called "secure delete")
+but instead delete the filesystem entries referring to the data. This leaves the random
+data on disk until the FS eventually claims it for another file. Therefore, if a hard
+drive nears capacity and files are deleted again, the change will barely decrease the
+space it takes up when compressed and deduplicated. Depending on the filesystem of the
+VM (or physical computer, if for some reason a normal filesystem backup can't be taken),
+there are several ways to decrease the size of a full image:
+
+Using ntfsclone (NTFS, i.e. Windows VMs)
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+ntfsclone can only operate on filesystems with the journal cleared (i.e. turned-off
+machines) which somewhat limits its utility in the case of VM snapshots. However,
+when it can be used, its special image format is even more efficient than just zeroing
+and deduplicating. For backup, save the MBR and the contents of each partition::
+
+    sfdisk -d /dev/sdX > sfdisk.txt
+    dd if=/dev/sdX of=mbr_gap count=2048  # all sectors until 1st partiton
+    borg create repo::hostname-partinfo-mbr-gap sfdisk.txt mbr_gap
+    sfdisk -l -o Device,Type /dev/sdX | grep NTFS | cut -d' ' -f1 | while read x; do
+        ntfsclone -s -o - $x | borg create repo::hostname-part$(echo $x | grep -Eo "[0-9]+$") -
+    done
```

Comment:
hmm, this reads a bit strange as this seems to be about a physical disk / machine and last thing you mentioned above was a VM.

maybe add a "check this!" in the comment of the dd, it depends on layout. there is no problem if we save a bit more than needed, but there might be a problem if we do not save the whole gap (if it is larger).

you practically tested the script?

Reply:
I tested it on a MBR-formatted loopback device with a few NTFS partitions, I am testing it with a full windows installation right now.

On second thought I think saving the MBR with `sfdisk` is kind of redundant, when copying all space leading up to the first partition should do virtually the same thing. I can parse that out of the `sfdisk` output (it seems to always be 2048 bytes on MBR disks). Maybe instead I should use `dd` to copy up to the first partition and then use `partprobe` to notify the kernel of the new partition layout instead of using `sfdisk` at all.

Should these instructions back up every partition, NTFS or not, using ntfsclone for ntfs ones? At that point this is becoming more full bash script territory instead of just a snippet.

Reply:
I just tested it on a Windows VM, which obviously used the MBR format. I added several partitions and checked restore, all the files were in the same places on the secondary NTFS & fat32 partitions and Windows booted (obviously I didn't meticulously check every windows system file but if it booted I'm assuming they stayed in place).